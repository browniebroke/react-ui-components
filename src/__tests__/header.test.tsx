import React from 'react'
import { ThemeProvider } from 'styled-components'
import renderer from 'react-test-renderer'

import { Header, styledComponentsTheme } from '../'

describe('Header component', () => {
  test('should render correctly with children', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Header>
            <div>This</div>
            <div>is</div>
            <div>cool</div>
          </Header>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly without children', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <Header />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
