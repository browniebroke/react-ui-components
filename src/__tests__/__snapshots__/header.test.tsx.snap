// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly with children 1`] = `
<header>
  <nav
    className="sc-eCssSg gDxVol"
  >
    <div
      className="sc-dlfnbm sc-jSgupP kMqpmM jGMqon"
    >
      <div>
        This
      </div>
      <div>
        is
      </div>
      <div>
        cool
      </div>
    </div>
  </nav>
</header>
`;

exports[`Header component should render correctly without children 1`] = `
<header>
  <nav
    className="sc-eCssSg gDxVol"
  >
    <div
      className="sc-dlfnbm sc-jSgupP kMqpmM jGMqon"
    />
  </nav>
</header>
`;
